File: 2. Structural Design Patterns/Facade/Go/FacadeTemplate.go
package main

// Facade Pattern – Go Template

type SubsystemA struct{}
func (SubsystemA) OpA(){ /* ... */ }

type SubsystemB struct{}
func (SubsystemB) OpB(){ /* ... */ }

type SubsystemC struct{}
func (SubsystemC) OpC(){ /* ... */ }

type Facade struct{ a SubsystemA; b SubsystemB; c SubsystemC }
func (f Facade) DoWork(){ f.a.OpA(); f.b.OpB(); f.c.OpC() }

func main(){ f := Facade{}; f.DoWork() }

File: 2. Structural Design Patterns/Facade/Java/FacadeTemplate.java
// Facade Pattern – Java Template

class SubsystemA { void opA(){ /* ... */ } }
class SubsystemB { void opB(){ /* ... */ } }
class SubsystemC { void opC(){ /* ... */ } }

class Facade { private final SubsystemA a = new SubsystemA(); private final SubsystemB b = new SubsystemB(); private final SubsystemC c = new SubsystemC(); void doWork(){ a.opA(); b.opB(); c.opC(); } }

public class FacadeTemplate { public static void main(String[] args){ new Facade().doWork(); } }
